tfx/components/example_gen/utils.py

```diff
@@ -209,84 +213,183 @@ def _glob_to_regex(glob_pattern: Text) -> Text:
   regex_pattern = regex_pattern.replace(')', '\\)')
   return regex_pattern
 
-
-def _retrieve_latest_span(uri: Text,
-                          split: example_gen_pb2.Input.Split) -> Text:
-  """Retrieves the most recently updated span matching a given split pattern."""
+def _glob_and_match_span_version(uri: Text,
+                                 split: example_gen_pb2.Input.Split,
+                                 is_match_version: bool
+                                 ) -> Tuple[Text, Optional[Text]]:
+  """Search for most recently updated properties given by `spec_list`."""
   split_pattern = os.path.join(uri, split.pattern)
-  if split_pattern.count(SPAN_SPEC) != 1:
-    raise ValueError('Only one {SPAN} is allowed in %s' % split_pattern)
 
-  split_glob_pattern = split_pattern.replace(SPAN_SPEC, '*')
+  split_glob_pattern = split_pattern
+  split_regex_pattern = _glob_to_regex(split_pattern)
+
+  split_glob_pattern = split_glob_pattern.replace(SPAN_SPEC, '*')
+  split_regex_pattern = split_regex_pattern.replace(SPAN_SPEC,
+      '(?P<{}>.*)'.format(SPAN_PROPERTY_NAME))
+
+  if is_match_version:
+    split_glob_pattern = split_glob_pattern.replace(VERSION_SPEC, '*')
+    split_regex_pattern = split_regex_pattern.replace(VERSION_SPEC,
+        '(?P<{}>.*)'.format(VERSION_PROPERTY_NAME))
+
   logging.info('Glob pattern for split %s: %s', split.name, split_glob_pattern)
-  split_regex_pattern = _glob_to_regex(split_pattern).replace(SPAN_SPEC, '(.*)')
   logging.info('Regex pattern for split %s: %s', split.name,
                split_regex_pattern)
-  if re.compile(split_regex_pattern).groups != 1:
-    raise ValueError('Regex should have only one group')
+
+  num_groups = re.compile(split_regex_pattern).groups
+  if is_match_version and num_groups != 2:
+    raise ValueError('Span and version regex should have two groups')
+  elif not is_match_version and num_groups != 1:
+    raise ValueError('Span only regex should have one group')
 
   files = tf.io.gfile.glob(split_glob_pattern)
   latest_span = None
+  latest_version = None
+
   for file_path in files:
     result = re.search(split_regex_pattern, file_path)
     if result is None:
       raise ValueError('Glob pattern does not match regex pattern')
+
+    select_span = result.group(SPAN_PROPERTY_NAME)
```

Comment:
consider call it 

latest_span/result_span, span_str and span_int as current iteration's span

latest_span, select_span, span feels confusing

Reply:
renamed